scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -629,29 +633,36 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
   @inline final def commaSeparated[T <: Tree: AstInfo](part: => T): List[T] =
     tokenSeparated[Comma, T](sepFirst = false, part)
 
-  private def makeTuple[T <: Tree](lpPos: Int, body: List[T], zero: => T, tuple: List[T] => T): T =
-    body match {
-      case Nil => autoEndPos(lpPos)(zero)
-      case only :: Nil =>
-        only match {
-          case q: Quasi if q.rank == 1 => copyPos(q)(tuple(body))
-          case _ => only
-        }
-      case _ => autoEndPos(lpPos)(tuple(body))
-    }
+  private def makeTuple[A <: Tree](lpPos: Int, body: List[A], zero: => A, tuple: List[A] => A)(
+      single: A => Either[List[A], A]
+  ): A = body match {
+    case Nil => autoEndPos(lpPos)(zero)
+    case (q: Quasi) :: Nil if q.rank == 1 => copyPos(q)(tuple(body))
+    case t :: Nil =>
+      single(t) match {
+        case Right(x) => x
+        case Left(x) => autoEndPos(lpPos)(tuple(x))
+      }
+    case _ => autoEndPos(lpPos)(tuple(body))
+  }
 
   def makeTupleTerm(lpPos: Int, body: List[Term]): Term = {
-    makeTuple(lpPos, body, Lit.Unit(), Term.Tuple.apply)
+    makeTuple(lpPos, body, Lit.Unit(), Term.Tuple.apply)(Right.apply)
   }
 
-  def makeTupleType(lpPos: Int, body: List[Type]): Type = {
-    val args = body match {
-      case singleArg :: Nil => maybeAnonymousLambda(singleArg) :: Nil
-      case _ => body
+  private def makeTupleType(lpPos: Int, body: List[Type], zero: => Type, wrap: Boolean): Type = {
+    makeTuple(lpPos, body, zero, Type.Tuple.apply) {
```

Comment:
Everywhere aside from here we just use t, so this is the only place we need to return anything aside from Boolean (so an Either), so it seems like we are complicating the signature of the `single` function just for this one case.

Reply:
what would you suggest? either we do it like this, or we duplicate the Quasi checks instead.